ipywidgets/widgets/widget_image.py

```diff
@@ -32,3 +44,66 @@ class Image(DOMWidget, ValueWidget, CoreWidget):
     width = CUnicode(help="Width of the image in pixels.").tag(sync=True)
     height = CUnicode(help="Height of the image in pixels.").tag(sync=True)
     value = Bytes(help="The image data as a byte string.").tag(sync=True)
+
+    @classmethod
+    def from_file(cls, filename, **kwargs):
+        """
+        Create an :class:`Image` from a local file.
+
+        Parameters
+        ----------
+        filename: str
+            The location of a file to read into the value from disk.
+
+        **kwargs:
+            The keyword arguments for `Image`
+
+        Returns an `Image` with the value set from the filename.
+        """
+        value = cls._load_file_value(filename)
+
+        return cls(value=value, **kwargs)
+
+    @classmethod
+    def from_url(cls, url, **kwargs):
+        """
+        Create an :class:`Image` from a URL.
+
+        :code:`Image.from_url(url)` is equivalent to:
+
+        .. code-block: python
+
+            img = Image(value=url, format='url')
+
+        But both unicode and bytes arguments are allowed for ``url``.
+
+        Parameters
+        ----------
+        url: [str, bytes]
+            The location of a URL to load.
+        """
+        if isinstance(url, _text_type):
+            url = url.encode('utf-8')
+
+        return cls(value=url, format='url')
+
+    def set_value_from_file(self, filename):
+        """
+        Convenience method for reading a file into `value`.
+
+        Parameters
+        ----------
+        filename: str
+            The location of a file to read into value from disk.
+        """
+        value = self._load_file_value(filename)
+
+        self.value = value
+
+    @classmethod
+    def _load_file_value(cls, filename):
+        if getattr(filename, 'read', None) is not None:
```

Comment:
Any reason not to use `if hasattr(filename, 'read')` ?

Reply:
Yes. It is mainly because [it's dangerous in Python 2](https://hynek.me/articles/hasattr/).

Reply:
Thanks, always learning!